core/common/bytestr.hpp

```diff
@@ -6,22 +6,28 @@
 
 #pragma once
 
-#include <libp2p/common/bytestr.hpp>
-#include <libp2p/common/span_size.hpp>
 #include <string_view>
 #include <vector>
 
+#include "common/buffer.hpp"
+#include "common/buffer_view.hpp"
+
 namespace kagome {
-  inline gsl::span<const uint8_t> str2byte(const gsl::span<const char> &s) {
-    return libp2p::bytestr(s);
+
+  inline common::BufferView str2byte(std::span<const char> s) {
+    // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+    return {reinterpret_cast<const uint8_t *>(s.data()), s.size()};
   }
 
-  inline std::string_view byte2str(const gsl::span<const uint8_t> &s) {
+  inline std::string_view byte2str(const common::BufferView &s) {
     // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
-    return {reinterpret_cast<const char *>(s.data()), libp2p::spanSize(s)};
+    return {reinterpret_cast<const char *>(s.data()), s.size()};
   }
 
-  inline std::vector<uint8_t> bytestr_copy(gsl::span<const char> s) {
-    return {s.begin(), s.end()};
+  inline common::Buffer bytestr_copy(std::string_view s) {
+    // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+    return common::BufferView{reinterpret_cast<const uint8_t *>(s.data()),
+                              s.size()};
```

Comment:
Unused, may be removed.
May be converted to `span2vec<T>(span<T>)` or `toVec<C>(C c) { vector{c.begin(), c.end()} }`, because vector constructor requires two arguments.
```suggestion
    return str2byte(s);
```